packages/doenetml-worker-rust/lib-rust-macros/src/component_attributes.rs

```diff
@@ -0,0 +1,173 @@
+use std::collections::HashMap;
+
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, TokenStream as TokenStream2};
+use quote::{format_ident, quote};
+use syn::{self, parse_macro_input, DeriveInput};
+
+use darling::{ast, util, FromDeriveInput, FromVariant};
+
+/// Struct for saving information about variants of the enum. This contains everything we need to
+/// generate code like
+/// ```ignore
+/// TextInputAttribute::Hide => BooleanStateVar::new_from_attribute("hide", false).into_state_var(),
+/// ```
+#[derive(Debug)]
+struct VariantImpl {
+    variant_name: String,
+    attr_name: String,
+    default: syn::Expr,
+    state_var: Ident,
+}
+
+impl VariantImpl {
+    /// Create a match case for this variant that looks like
+    /// ```ignore
+    /// TextInputAttribute::Hide => BooleanStateVar::new_from_attribute("hide", false).into_state_var(),
+    /// ```
+    fn to_match_case(&self, enum_name: &Ident) -> proc_macro2::TokenStream {
```

Comment:
I found the name of this method misleading, as the name made me think about matching the case of a string. Maybe `to_match_arm`?